sycl/doc/ControllingSYCLLanguageFeatures.md

```diff
@@ -0,0 +1,195 @@
+  text.
+
+  Note: setting `-sycl-std` option automatically implies a particular C++
+  standard version to be set.
+
+  **TODO**: do we need to specify which exact version is implied by which
+  `-sycl-std` value? AFAIK, SYCL spec only specifies minimum required C++
+  version, which is C++11 and we set C++17 in our implementation
+
+- `dpcpp-0.7`: corresponds to DPC++ 0.7.
+
+  Basically, `-sycl-std=dpcpp-0.7` implies support for SYCL 1.2.1 specification,
+  bunch of extension enabled and `-std=c++17`
+
+  See [oneAPI Specification] for more details and the specification text.
+
+[SYCL Registry]: https://www.khronos.org/registry/SYCL/
+[oneAPI Specification]: https://spec.oneapi.com/
+
+Note: it is possible to change C++ version independently of SYCL/DPC++ standard
+version if that is needed: `-sycl-std=1.2.1 -std=c++14`, for example.
+
+`-std=<value>`, `--std=<value>`, `--std <value>`
+
+One more way to specify SYCL/DPC++ standard version is to use general clang
+option which allows to specify language standard to compile for.
+
+Supported values (besides listed in clang documentation/help):
+
+- `sycl-1.2.1`, `sycl-2017`: corresponds to `-sycl-std=1.2.1`
+- `dpcpp-0.7`: corresponds to `-sycl-std=dpcpp-0.7`
+
+Please note that if you specify `-std` flag several times, only the latest
+value takes effect. This means, that if you want to specifiy particular C++
+standard version instead of some default one implied by SYCL/DPC++ standard,
+you have to use two separate options: `-sycl-std` and `-std`.
+
+**TODO**. Please let me know if we don't want to go with this approach at all.
+Note on motivation: clang accepts OpenCL standards via `-std` option,
+as well as `cuda` and `hip`. So, values mentioned above were added for
+consistency and for providing some additional aliases.
+
+## Controlling SYCL/DPC++ language extensions
+
+Both SYCL and DPC++ has several extensions or proposals about how to expand
+standard with new features, which might be vendor- or device-specific.
+
+Each extension can be separately enabled or disabled depending on user
+preferences. List of extensions enabled by default is controlled by SYCL/DPC++
+standard version.
+
+Enabling/disabling extensions is done via single `-sycl-ext` option. It accepts
+comma-separated list of extension names prefixed with `+` or `-` to indicate
+whether particular extension should be enabled or disabled.
+
+Example: `-sycl-ext=+EXTENSION_NAME1,-EXTENSION_NAME2` - this option specifies
+that `EXTENSION_NAME1` extension should be enabled and `EXTENSION_NAME2` should
+be disabled.
+
+When particular extension is enabled, the compiler automatically defines a macro
+with the same name, i.e. if `-sycl-ext=EXTENSION_NAME1` command line option was
+specified, then `EXTENSION_NAME1` macro will be defined.
+
+**TODO**: update table with supported extensions with identifiers (macro or
+compiler options), which should be used to enable/disable them.
```

Comment:
Sounds great, maybe the macro checks could read `__has_sycl_<extension>` for grouping/readability with other `__has_feature` and `__has_cpp_<feature>` options in the standard? https://en.cppreference.com/w/cpp/feature_test

+1 for a table to find flags, macros and reference.